engine.py

```diff
@@ -269,48 +357,106 @@ def show_panel(self, panel_id, title, bundle, widget_class, *args, **kwargs):
+                # Tell the user that they need to have the property
+                # pane present in the UI.
                 nuke.message("Cannot find any of the standard Nuke UI panels to anchor against. "
                              "Please add a Properties Bin to your Nuke UI layout and try again.")
                 return
     
-            # ok all good - we are running nuke 9 and/or 
+            # All good - we are running nuke 9 and/or 
             # have existing panes to parent against.
             panel_widget.addToPane(existing_pane)   
-        
         else:
-            # we are either calling this from a pane restore
+            # We are either calling this from a pane restore
             # callback or from the pane menu in Nuke. In both
             # these cases, the current pane is already established
             # by the system, so just add our widget.
-            # add it to the current pane
+            # Add it to the current pane
             panel_widget.addToPane()
+        return panel_widget
 
+    #####################################################################################
+    # Menu Utilities
+
+    def get_menu_selection(self):
+        """
+        Returns the list of hiero objects selected in the most recent menu click.
+        This list may contain items of various types. To see exactly what is being 
+        returned by which methods, turn on debug logging - this will print out details
+        of what is going on.
         
-        # return the created widget
-        return panel_widget     
+        Examples of types that are being returned are:
         
-    
-    ##########################################################################################
-    # managing favorite dirs            
+        Selecting a project in the bin view:
+        http://docs.thefoundry.co.uk/hiero/10/hieropythondevguide/api/api_core.html#hiero.core.Bin
+        
+        Selecting an item in a bin view:
+        http://docs.thefoundry.co.uk/hiero/10/hieropythondevguide/api/api_core.html#hiero.core.BinItem
+        
+        Selecting a track:
+        http://docs.thefoundry.co.uk/hiero/10/hieropythondevguide/api/api_core.html#hiero.core.TrackItem
+        """
+        return self._last_clicked_selection
+        
+    def get_menu_category(self):
+        """
+        Returns the UI area where the last menu click took place.
+        
+        Returns one of the following constants:
+        
+        - HieroEngine.HIERO_BIN_AREA
+        - HieroEngine.HIERO_SPREADSHEET_AREA
+        - HieroEngine.HIERO_TIMELINE_AREA
+        - None for unknown or undefined
+        """
+        return self._last_clicked_area
+
+    #####################################################################################
+    # General Utilities
+
+    @staticmethod
+    def set_project_root(event):
+        """Ensure any new projects get the project root or default startup 
+        projects get the project root set
```

Comment:
i think it would be useful to document the event parameter here in the doc string. It's not totally obvious what sort of thing gets passed in via that parameter.